python/ray/experimental/serve/task_runner.py

```diff
@@ -6,6 +6,7 @@
 from ray.experimental.serve import context as serve_context
 from ray.experimental.serve.context import FakeFlaskQuest, TaskContext
 from ray.experimental.serve.http_util import build_flask_request
+from collections import defaultdict
 
 
 class TaskRunner:
```

Comment:
I can see few ways to reduce the amount of repeated code in this section:
1. Move work_item -> args, kwargs transformation inside the work_item class. 
```python
args, kwargs = work_item.parse()
```
2. extract the invocation logic to separate function:
- `invoke_single`
- `invoke_batch`

3. raise error early. for example `Exception("Batched queries contain mixed context.")` can be raised earlier in the function.